### PR TITLE
Fix thematic map search spinner positioning

### PR DESCRIPTION
**Pull request by mikkomustikkamaa**
Unfortunately first solution in commit a763a34 did not work when search button was pushed by pressing enter key instead of clicking button with mouse. When button was pressed with enter, needed y- coordinate containing location of button (issuer of event) was not included in event. For this reason solution in commit b58dcae is implemented which works in both mentioned cases. In proposed solution spinner is placed 100 pixels above buttons in Get content of data search. With data source and indicator selections, spinner should be located as near these selections (works as before).

### CODE REVIEW

File: bundles/statistics/statsgrid2016/view/SearchFlyout.js [hunk 1]
```diff
@@ -66,28 +66,34 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.view.SearchFlyout', function (t
...
         btn.setTitle(this.loc('panels.newSearch.addButtonTitle'));
         btn.setEnabled(false);
-        btn.insertTo(container);
+        btn.insertTo(buttonContainer);
 
         btn.setHandler(function (event) {
             event.stopPropagation();
-            me.setSpinner(selectionComponent.spinner);
+            const progressSpinner = Oskari.clazz.create('Oskari.userinterface.component.ProgressSpinner');
+            progressSpinner.insertTo(jQuery('#statistics-search-flyout-button-container'));
```
Comment (ZakarFin) [inline bundles/statistics/statsgrid2016/view/SearchFlyout.js]: You can just pass the buttonContainer variable. Also remove the id as we try to use as little those as possible.

Comment (ZakarFin) [reply bundles/statistics/statsgrid2016/view/SearchFlyout.js]: Actually you can do the setup outside of the button handler and only start the spinner in the handler.

File: bundles/statistics/statsgrid2016/view/SearchFlyout.js [hunk 2]
```diff
@@ -66,28 +66,34 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.view.SearchFlyout', function (t
...
+            progressSpinner.opts.top = -100;
+            me.setSpinner(progressSpinner);
+            me.getSpinner().start();
```
Comment (ZakarFin) [inline bundles/statistics/statsgrid2016/view/SearchFlyout.js]: Nothing stops the spinner?

Comment (mikkomustikkamaa) [reply bundles/statistics/statsgrid2016/view/SearchFlyout.js]: Spinner is stopped at  https://github.com/mikkomustikkamaa/oskari-frontend/blob/b58dcaec61779b58dadfa54b49aa0a73248cf3b9/bundles/statistics/statsgrid2016/view/SearchFlyout.js#L522. Mentioned code in _addIndicators function was already implemented so i decided to use it.

File: bundles/statistics/statsgrid2016/view/SearchFlyout.js (hunk 1 above)
Comment (mikkomustikkamaa) [reply bundles/statistics/statsgrid2016/view/SearchFlyout.js]: In latest commit, spinner setup logic is moved from button event handler function. Also id attribute is removed from button container div.

**Pull request merged**